### PR TITLE
MG-2164 - Generate mocks with mockery for users

### LINKED ISSUE #2164: Generate mocks with mockery for Users service
### Is your feature request related to a problem? Please describe.
Yes, the feature request is related to the problem of testing the Users service. Currently, our tests are not isolated from the implementation details of the logger, which can lead to brittle tests.

### Describe the feature you are requesting, as well as the possible use case(s) for it.
We are requesting the feature of mock generation for the Users service. This feature will allow us to write unit tests that are isolated from the implementation details. The use case for this feature is primarily in testing - it will allow us to write more robust and reliable unit tests for the Users service.

### Indicate the importance of this feature to you.
Must-have

### PR DESCRIPTION
**Pull request by JeffMboya**
# What type of PR is this?

This is a feature because it adds mocks for Users service using mockery.

## What does this do?

This PR introduces mock generation for the Users service. This will allow us to write unit tests that are isolated from the implementation details of logger. 

## Which issue(s) does this PR fix/relate to?

- Related Issue #2164
- Resolves #2164

## Have you included tests for your changes?

Yes

## Did you document any new/modified feature?

No

### CODE REVIEW

File: users/emailer.go [hunk 1]
```diff
@@ -4,6 +4,8 @@
 package users
 
 // Emailer wrapper around the email.
+//
+//go:generate mockery --name Emailer --output=./mocks --filename email.go --quiet --note "Copyright (c) Abstract Machines"
```
Comment (rodneyosodo) [inline users/emailer.go]:
```suggestion
//go:generate mockery --name Emailer --output=./mocks --filename emailer.go --quiet --note "Copyright (c) Abstract Machines"
```

File: pkg/sdk/go/setup_test.go [hunk 2]
```diff
@@ -54,7 +54,7 @@ var (
 
 	subject   = generateUUID(&testing.T{})
 	object    = generateUUID(&testing.T{})
-	emailer   = umocks.NewEmailer()
+	emailer   = umocks.NewEmailer(&testing.T{})
```
Comment (rodneyosodo) [inline pkg/sdk/go/setup_test.go]:
```suggestion
	emailer   = new(mocks.Emailer)
```

**Review by rodneyosodo — CHANGES_REQUESTED** [review]

File: pkg/sdk/go/setup_test.go [hunk 3]
```diff
@@ -14,7 +14,7 @@ import (
 	sdk "github.com/absmach/magistrala/pkg/sdk/go"
 	"github.com/absmach/magistrala/pkg/uuid"
 	"github.com/absmach/magistrala/users/hasher"
-	umocks "github.com/absmach/magistrala/users/mocks"
+	"github.com/absmach/magistrala/users/mocks"
```
Comment (rodneyosodo) [inline pkg/sdk/go/setup_test.go]: Revert to using alias `umocks`

**Review by rodneyosodo — CHANGES_REQUESTED** [review]

File: users/emailer.go [hunk 4]
```diff
@@ -4,6 +4,8 @@
 package users
 
 // Emailer wrapper around the email.
+//
+//go:generate mockery --name Emailer --output=./mocks --filename emailer.go --quiet --note "Copyright (c) Abstract Machines"
```
Comment (rodneyosodo) [inline users/emailer.go:8]: Add this to CI

File: users/service_test.go [hunk 5]
```diff
@@ -2261,11 +2261,13 @@ func TestGenerateResetToken(t *testing.T) {
 	for _, tc := range cases {
 		repoCall := cRepo.On("RetrieveByIdentity", context.Background(), tc.email).Return(tc.retrieveByIdentityResponse, tc.retrieveByIdentityErr)
 		repoCall1 := auth.On("Issue", mock.Anything, mock.Anything).Return(tc.issueResponse, tc.issueErr)
+		repoCall2 := e.On("SendPasswordReset", mock.Anything, mock.Anything, mock.Anything, mock.Anything).Return(tc.err)
```
Comment (rodneyosodo) [inline users/service_test.go]: This is not a repocall. `repocall` is short form for repository call

**Review by rodneyosodo — CHANGES_REQUESTED** [review]

File: users/service_test.go [hunk 6]
```diff
@@ -2261,11 +2261,13 @@ func TestGenerateResetToken(t *testing.T) {
 	for _, tc := range cases {
 		repoCall := cRepo.On("RetrieveByIdentity", context.Background(), tc.email).Return(tc.retrieveByIdentityResponse, tc.retrieveByIdentityErr)
 		repoCall1 := auth.On("Issue", mock.Anything, mock.Anything).Return(tc.issueResponse, tc.issueErr)
+		svcCall := e.On("SendPasswordReset", mock.Anything, mock.Anything, mock.Anything, mock.Anything).Return(tc.err)
```
Comment (rodneyosodo) [inline users/service_test.go]: Reduce usage of `mock.Anything` if parameters to the function are known.

**Review by rodneyosodo — CHANGES_REQUESTED** [review]

**Review by rodneyosodo — APPROVED** [review]
LGTM

**Review by dborovcanin — APPROVED** [review]

**Pull request merged**